metricflow/model/model_transformer.py

```diff
@@ -37,25 +37,49 @@ class ModelTransformer:
         AddInputMetricMeasuresRule(),
     )
 
+    DEFAULT_RULES: Tuple[Sequence[ModelTransformRule], ...] = (
+        PRIMARY_RULES,
+        SECONDARY_RULES,
+    )
+
     @staticmethod
-    def pre_validation_transform_model(
-        model: UserConfiguredModel, rules: Sequence[ModelTransformRule] = DEFAULT_PRE_VALIDATION_RULES
+    def transform(
+        model: UserConfiguredModel,
+        ordered_rule_sequences: Tuple[Sequence[ModelTransformRule], ...] = DEFAULT_RULES,
```

Comment:
what's the benefit of having a tuple of sequences of rules rather than just a sequence of rules?

Reply:
There are a few benefits. one is of explicitness, one is of ease of ordering.

## Explicitness
Our transformation rules _require_ that some happen before or after one another. Specifically currently we have two rules which _must_ be run before any of our other current rules. There are two ways to do this. One way is a one dimension explicitly ordered list. I.e. a list wherein the first rule must be run before the second rule and so on. The other way is the way I've implemented in this PR, ranked/ordered groups of rules (primary rules, secondary rules, tertiary rules, and etc). I.e. primary rules have to be run before secondary rules, secondary rules before tertiary rules, and etc.

There is a difference in explicitness in the two solutions. With a flat ordered list you know the relation of one rule to the next. But how does a rule related to all of the other rules that come before and after it where is the dividing line? With an ordered sequence of non strictly ordered rule sequences it becomes very apparent. This plays into the second part, easy of updating

## Ease of updating
You create a new transform rule, where does it go? In the the flat ordered list you need to know where the dividing lines are of how rules relate. In the ordered sequence of non strictly ordered rule sequences it, the dividing lines are made apparent. Thus this implementation should make it easier for both future contributors and if people using MF as a package to add new/custom rules.